internal/backends/python/python.go

```diff
@@ -203,59 +202,24 @@ func makePythonPoetryBackend(python string) api.LanguageBackend {
 				return venv
 			}
 
-			// Ideally Poetry would provide some way of
-			// actually checking where the virtualenv will
-			// go. But it doesn't. So we have to
-			// reimplement the logic ourselves, which is
-			// totally fragile and disgusting. (No, we
-			// can't use 'poetry run which python' because
-			// that will *create* a virtualenv if one
-			// doesn't exist, and there's no workaround
-			// for that without mutating the global config
-			// file.)
-			//
-			// Note, we don't yet support Poetry's
-			// settings.virtualenvs.in-project. That would
-			// be a pretty easy fix, though. (Why is this
-			// so complicated??)
-
 			outputB, err := util.GetCmdOutputFallible([]string{
-				"poetry",
-				"config", "settings.virtualenvs.path",
+				"poetry", "env", "list", "--full-path",
 			})
 			if err != nil {
 				// there's no virtualenv configured, so no package directory
 				return ""
 			}
 
 			var path string
-			if err := json.Unmarshal(outputB, &path); err != nil {
-				util.Die("parsing output from Poetry: %s", err)
-			}
-
-			base := ""
-			if util.Exists("pyproject.toml") {
-				var cfg pyprojectTOML
-				if _, err := toml.DecodeFile("pyproject.toml", &cfg); err != nil {
-					util.Die("%s", err.Error())
+			for _, line := range strings.Split(strings.TrimSpace(string(outputB)), "\n") {
+				var isActive bool
+				path, isActive = strings.CutSuffix(line, " (Activated)")
+				if isActive {
+					break
 				}
-				base = cfg.Tool.Poetry.Name
 			}
 
-			if base == "" {
-				cwd, err := os.Getwd()
-				if err != nil {
-					util.Die("%s", err)
-				}
-				base = strings.ToLower(filepath.Base(cwd))
-			}
-
-			version := strings.TrimSpace(string(util.GetCmdOutput([]string{
-				python, "-c",
-				`import sys; print(".".join(map(str, sys.version_info[:2])))`,
-			})))
-
-			return filepath.Join(path, base+"-py"+version)
+			return path
```

Comment:
is this backwards-compatible? we need to be able to support several poetry versions - ideally *at least* the minimum of the current major version

Reply:
It is backwards compatible with the earliest 1.x version of poetry I can find on nixpkgs, 1.0.2. Unfortunately it is _not_ compatible with [0.12.16](https://github.com/python-poetry/poetry/releases/tag/0.12.16), though that's from May 2019, nearly 4 years old at this point. The reason I'm making this change is because the current functionality doesn't work at all for any 1.x version of poetry, to the best of my knowledge.

```
[nix-shell:/tmp/boop]# poetry add reqeuests
Creating virtualenv boop-s3yOBcnp-py3.7 in /root/.cache/pypoetry/virtualenvs

...

[nix-shell:/tmp/boop]# poetry env list --full-path
/root/.cache/pypoetry/virtualenvs/boop-s3yOBcnp-py3.7 (Activated)

[nix-shell:/tmp/boop]# poetry --version
Poetry version 1.0.2
```